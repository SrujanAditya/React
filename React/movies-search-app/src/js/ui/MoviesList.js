import React from 'react';
import MovieComp from './MovieComp';
import '../css/MovieList.css';


const MoviesList = ({movies,movieDes,history}) => (
    <div className="moviesContainer">
        {
            movies.map((movie, i) =>(
                <MovieComp key={i}
                        {...movie}
                        onClick={movieDes}
                        history = {history}
                        />
            )
        )}
    </div>
)

export default MoviesList;