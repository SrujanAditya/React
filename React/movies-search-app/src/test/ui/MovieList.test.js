import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import MoviesList from '../../js/ui/MoviesList';

Enzyme.configure({ adapter: new Adapter() })

describe('ui', () => {
    it('MoviesList component renders correctly', () => {
        const props = {
            movies : ['hi','hello'],
            movieDes : jest.fn(),
            history : {}
        }
        const movieListComp = shallow(<MoviesList {...props} />);
        expect(movieListComp).toMatchSnapshot();
    }) 
})
