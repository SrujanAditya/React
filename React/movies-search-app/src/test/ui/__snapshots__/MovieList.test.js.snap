// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ui MoviesList component renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MoviesList
    history={Object {}}
    movieDes={[MockFunction]}
    movies={
      Array [
        "hi",
        "hello",
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <MovieComp
          0="h"
          1="i"
          history={Object {}}
          onClick={[MockFunction]}
        />,
        <MovieComp
          0="h"
          1="e"
          2="l"
          3="l"
          4="o"
          history={Object {}}
          onClick={[MockFunction]}
        />,
      ],
      "className": "moviesContainer",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "function",
        "props": Object {
          "0": "h",
          "1": "i",
          "history": Object {},
          "onClick": [MockFunction],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "function",
        "props": Object {
          "0": "h",
          "1": "e",
          "2": "l",
          "3": "l",
          "4": "o",
          "history": Object {},
          "onClick": [MockFunction],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <MovieComp
            0="h"
            1="i"
            history={Object {}}
            onClick={[MockFunction]}
          />,
          <MovieComp
            0="h"
            1="e"
            2="l"
            3="l"
            4="o"
            history={Object {}}
            onClick={[MockFunction]}
          />,
        ],
        "className": "moviesContainer",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "function",
          "props": Object {
            "0": "h",
            "1": "i",
            "history": Object {},
            "onClick": [MockFunction],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "function",
          "props": Object {
            "0": "h",
            "1": "e",
            "2": "l",
            "3": "l",
            "4": "o",
            "history": Object {},
            "onClick": [MockFunction],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
